enhancements/console/dynamic-plugins.md

```diff
@@ -0,0 +1,369 @@
+creation-date: 2020-08-18
+last-updated: 2020-09-01
+status: implementable
+---
+
+# Dynamic Plugins for OpenShift Console
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+OpenShift Console currently has
+[static plugins](https://github.com/openshift/console/tree/master/frontend/packages/console-plugin-sdk)
+that allow teams to contribute features to the UI such as CNV and OCS. These plugins live in a
+[packages directory](https://github.com/openshift/console/tree/master/frontend/packages)
+inside the `openshift/console` repo and extend console through a well-defined
+plugin API. Static plugins are built with console and included in the console
+image. There are currently over a half-dozen plugins.
+
+In addition to static plugins, we'd like to introduce dynamic plugins. Dynamic
+plugins are contributed by operators and loaded at runtime. Eventually we want
+to migrate static plugins to dynamic plugins to decouple them from the console
+and allow plugins to be updated more frequently.
+
+## Motivation
+
+Static plugins have worked well, but they have some limitations:
+
+1. Static plugins can only be updated on the OpenShift release cadence. We want
+   to enable plugin teams to move faster.
+1. Static plugins are tied to the console version, not the operator version. We
+   want to be able to update plugins as operators change APIs or add features.
+   The right version of the plugin should be used for the installed version of
+   the operator.
+
+We already have a half-dozen teams contributing to console. We need to be able
+to scale as more teams contribute. This will be difficult if everyone is
+contributing to a single monorepo.
+
+### Goals
+
+* Operators can deliver console plugins separate from the console image and
+  update plugins when the operator updates.
+* The dynamic plugin API is similar to the static plugin API to ease migration.
+* Plugins can use shared console components such as list and details page components.
+* Shared components from core will be part of a well-defined plugin API.
+* Plugins can use [Patternfly 4](https://www.patternfly.org/v4/) components.
+* Cluster admins control what plugins are enabled.
+* Misbehaving plugins should not break console.
+* Existing static plugins are not affected and will continue to work as expected.
+
+### Non-Goals
+
+* Initially we don't plan to make this a public API. The target use is for Red
+  Hat operators. We might reevaluate later when dynamic plugins are more
+  mature.
+* We can't avoid breaking changes in console dependencies such as Patternfly
+  even if we don't break the console plugin API itself. We'll need a way for
+  plugins to declare compatibility.
```

Comment:
this isn't phrased as a non-goal.  Especially the part where you say "we'll need a way for plugins to declare compatibility"

the non-goal is "ensuring plugins continue to work when patternfly is upgraded"
but then i think you also need an explicit goal about how you will allow plugins to be disabled if they are not compatible w/ the patternfly version the console framework is providing.